packages/edit-post/src/plugins/index.js

```diff
@@ -15,21 +18,46 @@ import KeyboardShortcutsHelpMenuItem from './keyboard-shortcuts-help-menu-item';
 import ToolsMoreMenuGroup from '../components/header/tools-more-menu-group';
 import WelcomeGuideMenuItem from './welcome-guide-menu-item';
 
+function ManagePatternsMenuItem() {
+	const { isBlockTheme, isVisible } = useSelect( ( select ) => {
+		const { canUser } = select( coreStore );
+		const { getEditorSettings } = select( editorStore );
+
+		return {
+			// The site editor and templates both check whether the user has
+			// edit_theme_options capabilities. We can leverage that here and not
+			// display the manage patterns link if the user can't access it.
+			isVisible: canUser( 'read', 'templates' ),
+			isBlockTheme: getEditorSettings().__unstableIsBlockBasedTheme,
+		};
+	}, [] );
+
+	if ( ! isVisible ) {
+		return null;
+	}
```

Comment:
The `wp_block` post type doesn't require `edit_theme_options` capabilities. This will prevent users with lower caps from accessing Patterns when a site uses a classic theme.

We should probably avoid the early return and just use the fallback URL.

```
const url = isBlockTheme && canViewPatternLibrary ? patternsUrl : defaultUrl;
```

Reply:
Thanks, from other discussions today, I was under the impression users without those capabilities shouldn't have been able to edit reusable blocks and patterns.

I've made the suggested change so this PR remains more in line with how trunk works.